src/TestableHttpClient/IResponse.cs

```diff
@@ -0,0 +1,6 @@
+﻿namespace TestableHttpClient;
+
+public interface IResponse
+{
+    Task<HttpResponseMessage> GetResponseAsync(HttpRequestMessage requestMessage, CancellationToken cancellationToken);
```

Comment:
Consider changing this interface to `Task HandleResponse(HttpRequestContext context)` where the context contains the request, response and settings